system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -41,67 +43,136 @@ extern "C" MetricSink* getSinkInstance(const char *name, const IPropertyTree *pS
+{
+    StringBuffer hostDomain;
     StringBuffer hostProtocol;
     StringBuffer hostPort;
 
     Owned<IPropertyTree> pHostConfigTree = pSettingsTree->getPropTree("host");
     if (pHostConfigTree)
     {
-        pHostConfigTree->getProp("@name", hostName);
+        pHostConfigTree->getProp("@domain", hostDomain);
 
         if (!pHostConfigTree->getProp("@protocol", hostProtocol))
-        {
             hostProtocol.append("https");
-        }
 
         if (!pHostConfigTree->getProp("@port", hostPort))
-        {
             hostPort.append("9200");
-        }
     }
 
-    if (!hostName.isEmpty() && !hostPort.isEmpty() && !hostProtocol.isEmpty())
+    // Validate the host configuration minimal settings are present
+    if (hostDomain.isEmpty() || hostProtocol.isEmpty())
     {
-        elasticHostUrl.append(hostProtocol).append("://").append(hostName).append(":").append(hostPort);
+        WARNLOG("ElasticMetricSink: Host configuration missing domain and/or protocol");
+        return false;
     }
-    else
+
+    // build url for use with httplib Client
+    elasticHostUrl.append(hostProtocol).append("://").append(hostDomain);
+    if (!hostPort.isEmpty())
+        elasticHostUrl.append(":").append(hostPort);
+
+    // Read optional certificate file path
+    pHostConfigTree->getProp("@certificateFilePath", certificateFilePath);
+
+    // Get authentication settings, if present
+    Owned<IPropertyTree> pAuthConfigTree = pSettingsTree->getPropTree("authentication");
+    if (pAuthConfigTree)
     {
-        WARNLOG("ElasticMetricSink: Host configuration missing or invalid");
+        // Retrieve the authentication type and validate (only basic is supported)
+        if (!pAuthConfigTree->getProp("@type", authenticationType) || !streq(authenticationType, "basic"))
+        {
+            WARNLOG("ElasticMetricSink: Only basic authentication is supported");
+            return false;
+        }
+
+        StringBuffer credentialsSecretKey;
+        pAuthConfigTree->getProp("@credentialsSecret", credentialsSecretKey);  // vault/secrets key
+        if (!credentialsSecretKey.isEmpty())
+        {
+            StringBuffer credentialsVaultId;
+            pAuthConfigTree->getProp("@credentialsVaultId", credentialsVaultId);//optional HashiCorp vault ID
+
+            DBGLOG("Retrieving ElasticSearch host authentication username/password from secrets tree '%s', from vault '%s'",
+                   credentialsSecretKey.str(), !credentialsVaultId.isEmpty() ? credentialsVaultId.str() : "");
+
+            Owned<const IPropertyTree> secretTree(getSecret("authn", credentialsSecretKey.str(), credentialsVaultId, nullptr));
+            if (secretTree == nullptr)
+            {
+                WARNLOG("ElasticMetricSink: Unable to load secret tree '%s', from vault '%s'", credentialsSecretKey.str(),
+                        !credentialsVaultId.isEmpty() ? credentialsVaultId.str() : "n/a");
```

Comment:
the n/a seems fine, but it seems to differ from other similar cases where and empty string is provided in the output.

Reply:
IF the error is that the vault id is empty, the single quoted empty string is most appropriate.

Reply:
Absence of a vault ID is not an error. Absence of username and/or password is the error. The vault ID is informational to tell the user if attempting to retrieve from a vault (and which vault of course)

Reply:
ok, that seems more confusing to me. I wish we had a usability team to weigh in on these types of issues developers are typically not the best to judge.

Reply:
The API to retrieve the secret has an optional vault parameter. If present, the secret is retrieved from the vault. If absent, the secret is treated as a regular Kubernetes secret. Two different APIs would have been better. So, the vault is included in the log message to indicate if the retrieval was from a vault or not, 'n/a' in the latter case.